### PR TITLE
Test TokenSale

### PR DESCRIPTION
**Pull request by MrToph**
# Changelog

- [x] add tests for TokenSale
- [x] fix `test` script to run the tests

### CODE REVIEW

File: test/TokenSale.spec.ts [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+
+describe('TokenSale', async () => {
+  const [user, admin, saleRecipient, buyer1, buyer2, other] = waffle.provider.getWallets();
+  WHITELISTED_ACCOUNTS.push(buyer1.address, buyer2.address);
+
+  async function fixture() {
+    let TokenFactory = await ethers.getContractFactory('TestERC20');
+    let tokenIn = (await TokenFactory.connect(admin).deploy('USDC', 'USDC')) as IERC20;
+    let tokenOut = (await TokenFactory.connect(admin).deploy('ARENA', 'ARENA')) as IERC20;
+    let TokenLockFactory = await ethers.getContractFactory('TokenLock');
```
Comment (tabshaikh) [inline test/TokenSale.spec.ts:37]: instead of using `TokenLock` in deployment it would be better to use `RevokableTokenLock` contract

**Review by tabshaikh — CHANGES_REQUESTED** [review]

File: test/TokenSale.spec.ts (hunk 1 above)
Comment (MrToph) [reply test/TokenSale.spec.ts]: fixed

**Review by HickupHH3 — APPROVED** [review]
Can consider having a `shared/TimeManipulation.ts` script for time manipulation methods like `evm_setNextBlockTimestamp()` and `evm_increaseTime()`

**Comment by MrToph** [comment]
ok, moved it to a shared file

**Pull request merged**